File: P119 Pascal's Triangle II/Solution.java
class Solution {
    // Time complexity : O(rowIndex). Each term is calculated once, in constant time.
    // Space complexity : O(rowIndex). No extra space required other than that required to hold the output.
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new ArrayList();
        result.add(1);
        if(rowIndex == 0) {
            return result;
        }
        for(int i = 1; i < rowIndex; i++) {
            int element = (int) (result.get(i - 1) * (long) (rowIndex + 1 - i) / i);
            result.add(element);
        }
        result.add(1);
        return result;
    }
}

File: P119 Pascal's Triangle II/solution.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        result = []
        result = [None for _ in range(rowIndex+1)]
        result[0], result[-1] = 1, 1
        
        for j in range(1, rowIndex):
            result[j] = result[j-1] * (rowIndex + 1 - j) // j
        
        return result
